libtiledbvcf/src/stats/allele_count.cc

```diff
@@ -317,17 +327,38 @@ void AlleleCount::flush() {
         .set_offsets_buffer(COLUMN_NAME[GT], gt_offsets_)
         .set_data_buffer(COLUMN_NAME[COUNT], count_buffer_);
 
-    auto st = query_->submit();
+    query_->submit();
 
-    if (st != Query::Status::COMPLETE) {
-      LOG_FATAL("AlleleCount: error submitting TileDB write query");
+    if (remote_) {
+      if (query_->query_status() == Query::Status::FAILED) {
+        LOG_FATAL("AlleleCount: error submitting TileDB write query");
+      }
+    } else {
+      if (query_->query_status() != Query::Status::COMPLETE) {
+        LOG_FATAL("AlleleCount: error submitting TileDB write query");
+      }
     }
 
     // Insert sample names from this query into the set of fragment sample names
     fragment_sample_names_.insert(sample_names_.begin(), sample_names_.end());
     sample_names_.clear();
   }
 
+  // For remote global order writes, zero query buffers prior to
```

Comment:
Why are we zeroing buffers? The last submit call should be `submit_and_finalize`?